scripts/npm/check-node-modules.js

```diff
@@ -0,0 +1,175 @@
+// Implementation based on
+// https://github.com/angular/angular/blob/3b9c08676a4c921bbfa847802e08566fb601ba7a/tools/npm/check-node-modules.js
+'use strict';
+
+// Imports
+var fs = require('fs');
+var path = require('path');
+
+// Constants
+var NPM_SHRINKWRAP_FILE = 'npm-shrinkwrap.json';
+var NPM_SHRINKWRAP_CACHED_FILE = 'node_modules/npm-shrinkwrap.cached.json';
+var FS_OPTS = {encoding: 'utf-8'};
+var PROJECT_ROOT = path.join(__dirname, '../../');
+
+// Variables
+var progressIndicatorTimer = null;
+
+// Run
+_main();
+
+// Functions - Definitions
+function _main() {
+  var reinstallIfStale = process.argv.indexOf('--reinstall') !== -1;
+  checkNodeModules(reinstallIfStale);
+}
+
+function cacheNpmShrinkwrapFile() {
+  var absoluteMarkerFilePath = path.join(PROJECT_ROOT, NPM_SHRINKWRAP_FILE);
+  var absoluteCachedMarkerFilePath = path.join(PROJECT_ROOT, NPM_SHRINKWRAP_CACHED_FILE);
+
+  startProgressIndicator('    Caching marker file...');
+  copyFile(absoluteMarkerFilePath, absoluteCachedMarkerFilePath, onCopied);
+
+  // Helpers
+  function onCopied(err) {
+    stopProgressIndicator();
+    if (err) logError(err);
+  }
+}
+
+function checkNodeModules(reinstallIfStale) {
+  var nodeModulesOk = compareMarkerFiles();
+
+  if (nodeModulesOk) {
+    console.log(':-) npm dependencies are looking good!');
+  } else {
+    console.warn(':-( npm dependencies are stale or in an unknown state!');
```

Comment:
We ought to make this BIGGER and add a message about how to fix it. Something like:

```
console.warn('=====================================================================');
console.log(':-( npm dependencies are stale or in an unknown state!');
console.log('You can rebuild the dependencies by running `node scripts/npm/check-node-modules.js --reinstall`');
console.warn('=====================================================================');
```